pkg/metrics/metrics_test.go

```diff
@@ -315,38 +272,209 @@ func TestWriteMetric(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			t.Parallel()
 
+			var saveReq *http.Request
```

Comment:
How about just saving the body as a `[]byte` rather than saving the whole `*http.Request`? I think the body is the only part that is used/checked.